pkg/zk/generators.go

```diff
@@ -175,13 +174,54 @@ func makeZkPodSpec(z *v1beta1.ZookeeperCluster, volumes []v1.Volume) v1.PodSpec
 		Affinity:   z.Spec.Pod.Affinity,
 		Volumes:    append(z.Spec.Volumes, volumes...),
 	}
-	if reflect.DeepEqual(v1.PodSecurityContext{}, z.Spec.Pod.SecurityContext) {
+	if z.Spec.Pod.SecurityContext != nil {
 		podSpec.SecurityContext = z.Spec.Pod.SecurityContext
 	}
 	podSpec.NodeSelector = z.Spec.Pod.NodeSelector
 	podSpec.Tolerations = z.Spec.Pod.Tolerations
 	podSpec.TerminationGracePeriodSeconds = &z.Spec.Pod.TerminationGracePeriodSeconds
 	podSpec.ServiceAccountName = z.Spec.Pod.ServiceAccountName
+
+	if z.Spec.VolumePermissions == true && z.Spec.Pod.SecurityContext != nil {
```

Comment:
It sounds like a nit but I think that we should insert a warning after line 224, if VolumePermissions was set to true but no z.Spec.Pod.SecurityContext was supplied, it sounds like the user intended to use the new init container, but forgot to set the z.Spec.Pod.SecurityContext and so at the end of the day didn't get the full intended effect.

Logging it with a warning would bring attention to what is probably a misconfiguration of the Zookeeper CR they dropped in.

Reply:
Added log message